src/middleware/addUserContext.ts

```diff
@@ -44,13 +43,41 @@ const addUserContext = (
 		return;
 	}
 
-	selectOrCreateUser(stringToKeycloakId(keycloakUserId))
-		.then((user) => {
-			(req as AuthenticatedRequest).user = user;
-			next();
+	Promise.all([
```

Comment:
This entire function is nasty right now because we don't have `async` middleware and cannot use `await` as a result.

I started to refactor to use `async` but determined that it was a complex enough proposition to warrant its own issue and PR.  This is related to #700 which I think might be worth taking on sooner than later.